packages/panzer/disc-fe/src/Panzer_Evaluator_DomainInterface.cpp

```diff
@@ -0,0 +1,46 @@
+#include "Panzer_Evaluator_DomainInterface.hpp"
+#include "Panzer_Workset.hpp"
+#include "Teuchos_Assert.hpp"
+
+namespace panzer {
+
+  DomainEvaluator::DomainEvaluator(int domain) : domain_(domain) {}
+
+  void DomainEvaluator::setDomain(const int domain)
+  {domain_=domain;}
+
+  int DomainEvaluator::cellStartIndex(const panzer::Workset & workset) const
+  {
+    if (domain_ == ALL)
+      return 0;
+    else if (domain_ == OWNED)
+      return 0;
+    else if (domain_ == GHOST)
+      return workset.num_owned_cells();
+    else if (domain_ == REAL)
+      return 0;
+    else if (domain_ == VIRTUAL)
+      return workset.num_owned_cells() + workset.num_ghost_cells();
```

Comment:
This might be a good time to swap out the names for these functions for panzer consistent names. I think empire-fluid is the only code that uses them directly. They're only named this to be consistent with 'num_cells'.